core/rpc/src/tests.rs

```diff
@@ -92,14 +92,14 @@ impl TestNode {
     fn query_runner(&self) -> fdi::Ref<QueryRunner> {
         self.inner.provider.get()
     }
-    fn blockstore(&self) -> fdi::Ref<Blockstore<TestBinding>> {
-        self.inner.provider.get()
-    }
 }
 
 async fn init_rpc(temp_dir: &TempDir, genesis_path: ResolvedPathBuf, rpc_port: u16) -> TestNode {
     let app_config = AppConfig::test(genesis_path);
 
+    let mut rpc_config = RpcConfig::default_with_port(rpc_port);
+    rpc_config.hmac_secret = Some(temp_dir.path().to_path_buf());
```

Comment:
@nhtyy I think you can fix the failing test by injecting this into the config below in the node `init_with_provider` call:
```rust
                .with::<Rpc<TestBinding>>(RpcConfig{
                    hmac_secret: Some(temp_dir.path().to_path_buf()),
                    ..RpcConfig::default_with_port(rpc_port)
                })
```

Instead of just

```rust
                .with::<Rpc<TestBinding>>(RpcConfig::default_with_port(rpc_port))
```

Reply:
didnt notice it was never used thanks 😓